frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx

```diff
@@ -238,9 +243,12 @@ const CloudShellExec: React.FC<CloudShellExecProps> = ({
 
   if (wsOpen) {
     return (
-      <div className="co-cloudshell-terminal__container">
-        <Terminal onData={onData} ref={terminal} />
-      </div>
+      <>
+        <div className="co-cloudshell-terminal__container">
+          <Terminal onData={onData} ref={terminal} />
+        </div>
+        <ExecuteCommand onCommand={onCommand} />
```

Comment:
Do we really need a separate component here to run the command?

Reply:
We can add the logic to execute command in the CloudShellExec component itself but every time a command is sent the component will rerender and would require extra checks(like web socket is open or not) to execute commands. This way when command is sent only the Execute component is affected and will only render when websocket is open. 🙂

Reply:
Okay, yeah. It makes sense to keep this logic to prevent re-renders of parent component.